src/main/java/seedu/address/model/animal/Task.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.model.animal;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents the Task addressed to the {@link Animal}
+ *
+ */
+public class Task {
+
+    public static final String MESSAGE_CONSRAINTS = "Description should not be empty!";
+    public static final String VALIDATION_REGEX = "^.*\\S.*$";
+
+    public final boolean isDone;
+    public final String description;
+
+
+    /**
+     * Constructs a Task Object. isDone status of tasks are initialized to be false.
+     *
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        requireNonNull(description);
+        checkArgument(isValidTask(description), MESSAGE_CONSRAINTS);
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a Task Object. This method is only used as the constructor when converting
+     * from a Jackson-friendly adapted Task to an actual Task Object.
+     *
+     * @param description Description of the task.
+     * @param isDone isDone status of the task.
+     */
+    public Task(String description, boolean isDone) {
+        requireNonNull(description);
+        checkArgument(isValidTask(description), MESSAGE_CONSRAINTS);
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return The description of the task.
+     */
+    public String getDescription() {
+        return this.description;
```

Comment:
For the following methods, `this` is not required to disambiguate the use of attributes. This is also to keep the style in line with the rest of the code.